src/ai/backend/test/testcases/session/execution_failure_wrong_command.py

```diff
@@ -0,0 +1,52 @@
+import asyncio
+
+from ai.backend.client.exceptions import BackendAPIError
+from ai.backend.common.types import ClusterMode
+from ai.backend.test.contexts.client_session import ClientSessionContext
+from ai.backend.test.contexts.image import ImageContext
+from ai.backend.test.contexts.session import SessionContext
+from ai.backend.test.contexts.sse import (
+    SSEContext,
+)
+from ai.backend.test.templates.template import TestCode
+
+
+class BatchSessionCreationFailureWrongCommand(TestCode):
+    async def test(self) -> None:
+        client_session = ClientSessionContext.current()
+        image_dep = ImageContext.current()
+        sse_dep = SSEContext.current()
+        session_dep = SessionContext.current()
+
+        session_name = "test-batch-session-execution-failure"
+
+        async def collect_events():
+            async with client_session.ComputeSession(session_name).listen_events() as events:
+                async for event in events:
+                    if event.event == "session_failure":
+                        break
+                    if event.event == "session_cancelled":
+                        raise RuntimeError("BatchSession was cancelled")
+                    if event.event == "session_success":
+                        raise RuntimeError("BatchSession should not succeed with wrong command")
+
+        listener_task = asyncio.create_task(
+            asyncio.wait_for(collect_events(), timeout=sse_dep.timeout)
+        )
+
+        try:
+            await client_session.ComputeSession.get_or_create(
+                image_dep.name,
+                architecture=image_dep.architecture,
+                name=session_name,
+                type_="batch",
+                startup_command="some_wrong_command!@#123",
+                resources=session_dep.resources,
+                cluster_mode=ClusterMode.SINGLE_NODE,
+                cluster_size=1,
+            )
+
+            await listener_task
+        except BackendAPIError:
+            # TODO: Should `get_or_create` raise error when the command fails?
```

Comment:
I'm not sure if we really need to except exceptions. I think it might not convey information as well.